applications/IgaApplication/custom_conditions/support_laplacian_condition.cpp

```diff
@@ -0,0 +1,157 @@
+
+        normal_physical_space = normal_parameter_space; // prod(trans(J0[0]),normal_parameter_space);
+        
+        for (IndexType point_number = 0; point_number < integration_points.size(); ++point_number)
+        {
+            const Matrix& N = r_geometry.ShapeFunctionsValues();
+
+            // Calculating the cartesian derivatives
+            noalias(DN_DX) = DN_De[point_number]; // prod(DN_De[point_number],InvJ0);
+            
+            Matrix H = ZeroMatrix(1, number_of_nodes);
+            Matrix DN_dot_n = ZeroMatrix(1, number_of_nodes);
+            Vector DN_dot_n_vec = ZeroVector(number_of_nodes);
+            Vector H_vector = ZeroVector(number_of_nodes);
+            for (IndexType i = 0; i < number_of_nodes; ++i)
+            {
+                H(0, i)            = N(point_number, i);
+                H_vector(i)        = N(point_number, i); 
+
+                for (unsigned int idim = 0; idim < dim; idim++) {
+                    DN_dot_n(0, i)   += DN_DX(i, idim) * normal_physical_space[idim];           
+                    DN_dot_n_vec(i)  += DN_DX(i, idim) * normal_physical_space[idim];
+                } 
+                
+            }
+            // Differential area
+            double penalty_integration = penalty * integration_points[point_number].Weight(); //  * std::abs(DetJ0);
+
+            // Collins, Lozinsky & Scovazzi innovation
+            double Guglielmo_innovation = 1.0;  // = 1 -> Penalty approach
+                                                // = -1 -> Free-penalty approach
+            if (penalty == -1.0) {
+                penalty_integration = 0.0;
+                Guglielmo_innovation = -1.0;
+            }
+            
+
+            // Assembly
+            noalias(rLeftHandSideMatrix) -= prod(trans(H), H) * penalty_integration;
+            // Assembly of the integration by parts term -(w,GRAD_u * n) -> Fundamental !!
+            noalias(rLeftHandSideMatrix) -= prod(trans(H), DN_dot_n)                        * integration_points[point_number].Weight(); // * std::abs(DetJ0) ;
+            // Assembly Dirichlet BCs -(GRAD_w* n,u) 
+            noalias(rLeftHandSideMatrix) -= Guglielmo_innovation * prod(trans(DN_dot_n), H) * integration_points[point_number].Weight(); // * std::abs(DetJ0) ;
+
+
+            if (CalculateResidualVectorFlag) {
+                
+                const double u_D_scalar = this->GetValue(TEMPERATURE);
+
+                noalias(rRightHandSideVector) -=  H_vector * u_D_scalar * penalty_integration;
+                noalias(rRightHandSideVector) -= Guglielmo_innovation * DN_dot_n_vec * u_D_scalar * integration_points[point_number].Weight(); // * std::abs(DetJ0);
+
+                Vector temp(number_of_nodes);
+                // RHS = ExtForces - K*temp;
+                for (IndexType i = 0; i < number_of_nodes; i++) {
+                    temp[i] = r_geometry[i].GetSolutionStepValue(TEMPERATURE);

```

Comment:
Shouldn't you take this as the unknown variable from the convection diffusion settings?